src/targets/gpu/jit/mlir.cpp

```diff
@@ -74,6 +74,57 @@ struct mlir_compiler : compiler<mlir_compiler>
 
     operation compile_op(context&, const std::vector<shape>&, const value&) const { return {}; }
 
+    const auto& reshaper_names() const
+    {
+        // clang-format off
+        static const std::unordered_set<std::string> names = {
+            "slice",
+            "transpose",
+            "multibroadcast",
+            "broadcast",
+            "contiguous",
+            "reshape",
+            "lazy_reshape",
+            "squeeze",
+            "flatten",
+            "unsqueeze"
+        };
+        // clang-format on
+        return names;
+    }
+
+    std::tuple<bool, instruction_ref> input_is_param(const instruction_ref& ins) const
+    {
+        auto cur = ins;
+        while(contains(reshaper_names(), cur->name()))
+        {
+            cur = cur->inputs().at(0);
+        }
+
+        return {cur->name() == "@param", cur};
+    }
+
+    void set_fill_map(compiler_replace& cr, const module& m) const
+    {
+        for(auto ins : iterator_for(m))
+        {
+            if(ins->name() == "greater")
+            {
+                auto fill_val = ins->get_shape().lens().back() - 1;
+                for(auto inp : ins->inputs())
```

Comment:
You need to check that one input is a range literal and the other is an input param.